auto_submit/lib/requests/check_pull_request.dart

```diff
@@ -39,14 +45,62 @@ class CheckPullRequest extends RequestHandler {
       return responses;
     }
     final List<Future<Response>> futures = <Future<Response>>[];
+    Set shouldMergeSet = <PullRequest>{};
     for (pub.ReceivedMessage message in receivedMessages) {
-      futures.add(_processMessage(message));
+      futures.add(_processMessage(message, shouldMergeSet));
     }
     responses.addAll(await Future.wait(futures));
+
+    final Set mergeSet = await checkMerge(shouldMergeSet);
+    await mergePullRequest(mergeSet);
+    return responses;
+  }
+
+  /// Check and collect the pull requests to merge this cycle.
+  ///
+  /// The number of pull requests to be merged to each repo will not exceed
+  /// the _kMergeCountPerRepo
+  Future<Set> checkMerge(Set shouldMergeSet) async {
+    //The repo map stores the repo name and the count of PRs to merge to this repo
+    HashMap<String, int> repos = HashMap<String, int>();
+    Set mergeSet = HashSet<PullRequest>();
+    for (PullRequest pr in shouldMergeSet) {
+      String repoName = pr.base!.repo!.slug().fullName;
+      if (!repos.containsKey(repoName) || repos[repoName]! < _kMergeCountPerRepo) {
+        repos[repoName] = repos.containsKey(repoName) ? repos[repoName]! + 1 : 1;
+        mergeSet.add(pr);
```

Comment:
Can we directly merge the PR here? I don't see why we need to collect the PR set and then process the PR iterating through the set.